sm-shop/src/main/java/com/salesmanager/shop/store/api/v0/base/BaseClass.java

```diff
@@ -0,0 +1,39 @@
+package com.salesmanager.shop.store.api.v0.base;
+
+import java.io.IOException;
+
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletResponse;
+
+import com.salesmanager.core.business.exception.ServiceException;
+import com.salesmanager.core.business.services.merchant.MerchantStoreService;
+import com.salesmanager.core.model.merchant.MerchantStore;
+
+public class BaseClass {
+	
+	@Inject
+	private MerchantStoreService merchantStoreService;
+	
+	/**
+	 * @param store
+	 * @param response
+	 * @param merchantStore
+	 * @return
+	 * @throws ServiceException
+	 * @throws IOException
+	 */
+	protected MerchantStore getMerchantStore(final String store, HttpServletResponse response, MerchantStore merchantStore)
```

Comment:
Using inheritance in this case is not appropriate, choose relation 'HAS-A' over the relation 'IS -A' because inheritance is lot expensive.  I understand your code factorisation but   In my opinion the service should be called from the controller and it's the appropriate way, making a parent class and call it from all the controller is not clean. Let the controller without inheritance it's better .
I recommend you not merge  and let the code as it was

Reply:
Hi @taarh 

I am about to decommission many Controllers in master that will conflict with the changes submitted
Can you put your PR in https://github.com/shopizer-ecommerce/shopizer/tree/2.17.0
the last version to support controllers

Reply:
Hi @shopizer-ecommerce 

Are you asking me to raise PR in https://github.com/shopizer-ecommerce/shopizer/tree/2.17.0.

Reply:
Hi @shopizer-ecommerce   that's sound good but i have another solution instead of making AbstractController, you could transmform it as SessionUtility and for  your pagination mae it PaginationUtility.  dont force your controller to extends classes if you have alternatif.